macOS wheels don't work on macOS 10.12

The wheels report 10.9+ compatibility but don't work on 10.12. The same error happens for the 1.2.0 and 1.2.1 wheels:

```
Python 3.7.1 (default, Apr  7 2020, 18:00:22)
[Clang 9.0.0 (clang-900.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pygit2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "...lib/python3.7/site-packages/pygit2/__init__.py", line 30, in <module>
    from ._pygit2 import *
ImportError: dlopen(...lib/python3.7/site-packages/pygit2/_pygit2.cpython-37m-darwin.so, 2): Symbol not found: ____chkstk_darwin
  Referenced from: ...lib/python3.7/site-packages/pygit2/.dylibs/libgit2.1.0.0.dylib
  Expected in: /usr/lib/libSystem.B.dylib
 in ...lib/python3.7/site-packages/pygit2/.dylibs/libgit2.1.0.0.dylib
```

I guess they use a symbol from a newer libSystem.

(I know the workaround is to do `pip install --no-binary pygit2 pygit2`).

In case it's related, see https://github.com/pyca/cryptography/issues/5215 and https://github.com/pyca/infra/pull/275

> The wheel you linked to works!

That seems suspiciously easy... 🕵️ 

There are some warnings during the [libgit2 compilation](https://github.com/rcoup/pygit2/actions/runs/197670928):

```
/Users/runner/work/pygit2/pygit2/libgit2/src/unix/posix.h:86:9: warning: 'futimens' is only available on macOS 10.13 or newer [-Wunguarded-availability-new]
        return futimens(f, s);
               ^~~~~~~~
/Applications/Xcode_11.6.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include/sys/stat.h:396:9: note: 'futimens' has been marked as being introduced in macOS 10.13 here, but the deployment target is macOS 10.9.0
int     futimens(int __fd, const struct timespec __times[2]) __API_AVAILABLE(macosx(10.13), ios(11.0), tvos(11.0), watchos(4.0));
        ^
/Users/runner/work/pygit2/pygit2/libgit2/src/unix/posix.h:86:9: note: enclose 'futimens' in a __builtin_available check to silence this warning
        return futimens(f, s);
               ^~~~~~~~
1 warning generated.
```

Seems like `futimens()` is only [used in win32 though](https://github.com/libgit2/libgit2/search?q=p_futimes&type=Code), so might be safe enough to ignore.

Libgit2 cmake actually checks for it explicitly:
```
-- Looking for futimens
-- Looking for futimens - found
```
though https://gitlab.kitware.com/cmake/cmake/-/issues/17101#note_297983 suggests the CheckFunctionExists test in cmake doesn't actually respect macOS deployment targets.

@jdavid happy to commit this change anyway?

--- a/.github/workflows/macos.yml
+++ b/.github/workflows/macos.yml
@@ -24,6 +24,7 @@ jobs:
       CCACHE_DIR: ${{ github.workspace }}/.cache/ccache
       CCACHE_COMPRESS: 1
       PIP_CACHE_DIR: ${{ github.workspace }}/.cache/pip
+      MACOSX_DEPLOYMENT_TARGET: "10.9"
     steps:
       - name: checkout pygit2
         uses: actions/checkout@v2
@@ -79,6 +80,7 @@ jobs:
           -DBUILD_CLAR=NO \
           -DUSE_SSH=NO
           cmake --build . --target install
+          otool -L "${{ github.workspace }}/libgit2/env/lib/libgit2.dylib"
           VERSION=$(PKG_CONFIG_PATH=$(pwd) pkg-config --modversion libgit2)
           echo "::set-output name=version::$VERSION"
 

